grails-forge-core/src/main/java/org/grails/forge/feature/github/workflows/plain/templates/plainGithubWorkflow.rocker.raw

```diff
@@ -12,23 +12,24 @@ on: [push, pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      contents: read
     steps:
 @javaSetup.template(jdkVersion)
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@@v3
       - name: Run Tests
         if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
         id: tests
-        uses: gradle/gradle-build-action@@v2
-        with:
-          arguments: check
+        run: ./gradlew check
       - name: Run Build
         if: github.event_name == 'push'
         id: build
-        uses: gradle/gradle-build-action@@v2
-        with:
-          arguments: build
+        run: ./gradlew build
       - name: Publish Test Report
-        if: steps.build.outcome == 'failure' || steps.tests.outcome == 'failure'
+        if: failure()
```

Comment:
The Publish Test Report step was not executing for me when my `./gradlew build` step was failing. Changing this condition to `failure()` fixed that issue.